lib/go/thrift/debug_protocol.go

```diff
@@ -21,247 +21,269 @@ package thrift
 
 import (
 	"context"
-	"log"
+	"fmt"
 )
 
 type TDebugProtocol struct {
 	Delegate  TProtocol
 	LogPrefix string
+	Logger    Logger
 }
 
 type TDebugProtocolFactory struct {
 	Underlying TProtocolFactory
 	LogPrefix  string
+	Logger     Logger
 }
 
+// NewTDebugProtocolFactory creates a TDebugProtocolFactory.
+//
+// Deprecated. Please use NewTDebugProtocolFactoryWithLogger or the struct
```

Comment:
Should be `Deprecated: Please...` to be detectable by IDEs and static analysis tools (see [here](https://github.com/golang/go/wiki/Deprecated)).

Reply:
Done.